### PR TITLE
Administration system

### PR DESCRIPTION
**Pull request by VitaliyKovtun**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -42,6 +42,12 @@ gem 'font-awesome-rails'
 # Pagination library for Rails
 gem 'will_paginate'
 
+# OmniAuth is a library that standardizes multi-provider authentication
+gem 'omniauth'
```
Comment (pavel-mihailovskiy) [inline Gemfile]: why do you need this?

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -1,26 +1,29 @@
...
-    def comment_params
-      params.require(:comment).permit(:name, :body).merge(user_id: current_user.id)
-    end
+  def comment_params
+    params.require(:comment).permit(:name, :body).merge(user_id: current_user.id)
+  end
 
+  def find_comment
+    @post = Post.find(params[:post_id])
+    render text: 'Access denied', status: 403 unless @post
```
Comment (pavel-mihailovskiy) [inline app/controllers/comments_controller.rb]: i can't understand why you use 'Access denied' if record is not found.

File: app/models/comment.rb [hunk 3]
```diff
@@ -1,6 +1,7 @@
 class Comment < ActiveRecord::Base
   belongs_to :post
   belongs_to :user
+  belongs_to :admin
```
Comment (pavel-mihailovskiy) [inline app/models/comment.rb]: wtf??!

File: app/views/comments/_comment.html.erb [hunk 4]
```diff
@@ -1,12 +1,11 @@
...
-    <p class="comment_time"><%= time_ago_in_words(comment.created_at) %> by <%= comment.user.name %></p>
+<div class='comment clearfix'>
+  <div class='comment_content'>
+    <p class='comment_name'><strong><%= comment.name %></strong></p>
+    <p class='comment_body'><%= comment.body %></p>
+    <p class='comment_time'><%= time_ago_in_words(comment.created_at) %> by <%= comment.user.name %></p>
   </div>
 
-  <% if comment.user == current_user %>
+  <% if comment.user == current_user || current_user.admin? %>
```
Comment (pavel-mihailovskiy) [inline app/views/comments/_comment.html.erb]: It should be moved to helper or presenter

File: app/views/posts/index.html.erb [hunk 5]
```diff
@@ -1,11 +1,11 @@
...
+  <p class='image'><%= image_tag post.image.url(:thumb) %></p>
+  <div class='post_wrapper clearfix'>
+    <h4 class='title'><%= link_to post.title, post %></h4>
+    <p class='date'><%= post.created_at.strftime('%B, %d, %Y') %></p>
   </div>
 <% end %>
 
-<div class="pagination">
+<div class='pagination'>
   <%= will_paginate @posts, inner_window: 3, outer_window: 0, container: false %>
```
Comment (pavel-mihailovskiy) [inline app/views/posts/index.html.erb]: Why do you use will_paginate and kaminari? use one of them

File: Gemfile [hunk 6]
```diff
@@ -42,6 +42,12 @@ gem 'font-awesome-rails'
...
+
+# RailsAdmin is a Rails engine that provides an easy-to-use interface for managing your data
+gem 'rails_admin'
```
Comment (pavel-mihailovskiy) [inline Gemfile]: I asked not to use this gem

File: Gemfile (hunk 1 above)
Comment (VitaliyKovtun) [inline Gemfile]: done

File: app/controllers/comments_controller.rb (hunk 2 above)
Comment (VitaliyKovtun) [inline app/controllers/comments_controller.rb]: done

File: app/models/comment.rb (hunk 3 above)
Comment (VitaliyKovtun) [inline app/models/comment.rb]: done

File: app/views/posts/index.html.erb (hunk 5 above)
Comment (VitaliyKovtun) [inline app/views/posts/index.html.erb]: done

File: app/views/comments/_comment.html.erb (hunk 4 above)
Comment (VitaliyKovtun) [inline app/views/comments/_comment.html.erb]: done

**Pull request merged**